src/Databases/DatabaseOnDisk.cpp

```diff
@@ -368,7 +368,7 @@ void DatabaseOnDisk::dropTable(ContextPtr local_context, const String & table_na
         // The table might be not loaded for Lazy database engine.
         if (table)
         {
-            table->drop();
+            table->drop(local_context);
```

Comment:
Sometimes if it is difficult to pass context through the code - you can use `CurrentThread::getQueryContext()` to get local context from anywhere.